filesystem/replicator_test.go

```diff
@@ -0,0 +1,121 @@
+package filesystem
+
+import (
+	"os"
+	"testing"
+	"time"
+)
+
+func TestReplicateProcessFileCallBack(t *testing.T) {
+	t.Run("NonExistentSourceFile", func(t *testing.T) {
+		sourceFilePath := "non_existent_file.txt"
+		destinationFilePath := "destination.txt"
+		config := struct{}{}
+		err := replicateProcessFileCallBack(sourceFilePath, destinationFilePath, config)
+		if err == nil {
+			t.Errorf("Expected error for non-existent source file, got nil")
+		}
+	})
+
+	t.Run("CopySourceToDestination", func(t *testing.T) {
+		sourceFile, err := os.Create("source_file.txt")
```

Comment:
Instead of creating an empty file we can put a file under `testdata` folder and use that. It also mean we don't need to delete the file, so can remove the `defer os.Remove()`